src/main/java/de/tum/in/www1/artemis/service/iris/session/AbstractIrisChatSessionService.java

```diff
@@ -0,0 +1,34 @@
+package de.tum.in.www1.artemis.service.iris.session;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import de.tum.in.www1.artemis.domain.iris.session.IrisChatSession;
+import de.tum.in.www1.artemis.repository.iris.IrisSessionRepository;
+
+public abstract class AbstractIrisChatSessionService<S extends IrisChatSession> implements IrisChatBasedFeatureInterface<S> {
```

Comment:
```suggestion
public abstract class AbstractIrisChatSessionService<S extends IrisChatSession> implements IrisChatBasedFeatureInterface<S>, IrisRateLimitedFeatureInterface {
```
Can we already inherit from the interface in this super class?

Reply:
Implemented the suggested change. Thanks for reviewing :)